lib/Hakyll/Web/Template/Context.hs

```diff
@@ -151,15 +247,67 @@ mapContext f (Context c) = Context $ \k a i -> do
 -- > $snippet("path/to/snippet/")$
 -- > ...
 --
--- The contents of the included file will not be interpolated.
+-- The contents of the included file will not be interpolated like @partial@
+-- does it.
 --
 snippetField :: Context String
 snippetField = functionField "snippet" f
   where
     f [contentsPath] _ = loadBody (fromFilePath contentsPath)
-    f _              i = error $
-        "Too many arguments to function 'snippet()' in item " ++
-            show (itemIdentifier i)
+    f []             _ = fail "No argument to function 'snippet()'"
+    f _              _ = fail "Too many arguments to function 'snippet()'"
+
+
+
+dataField :: String -> Value -> Context a
+dataField key val = Context $ \f a (Item i _) -> case splitAll "." f of
+    [k] | k == get    -> lookupNestedValue a  (Item i val)
+    (k:ks) | k == key -> lookupNestedValue ks (Item i val)
+    _                 -> failBranch $ "Tried field " ++ key -- and functionField get
+  where
+    get = let (h:rest) = key in "get" ++ toUpper h : rest
```

Comment:
I don't really understand what this does :thinking: 

Are we encoding `getFoo` as special keys?

Reply:
This allows one to create a `dataField "foo" (Object metadata)` or something, then use `$foo.bar.baz$` in templates or `$getFoo("bar", "baz")$` - this allows lookup with dynamic names (generated by another field) and also supports names with dots (or any other special syntax).